pixeltable/share/packager.py

```diff
@@ -104,86 +110,116 @@ def __export_table(self, t: catalog.Table) -> None:
         # to get the column types, since we'll be substituting `fileurl`s for media columns.
         actual_col_types: list[ts.ColumnType] = []
 
-        for col_name, col in t._tbl_version.get().cols_by_name.items():
+        tv = tvp.tbl_version.get()
```

Comment:
This seems error-prone and a lot more complicated than necessary.

Why not simply have a
StoreBase.dump_rows(version, filter_view: StoreBase, filter_view_version: int) -> Iterator[dict]
that produces table rows in the same format as what's needed for sql.insert()

And a corresponding StoreBase.load_rows(rows: Iterator[dict]). That could then also do its own batching.

StoreBase already understands the physical structure of the table, and you're replicating the physical structure (ie, you shouldn't need any back & forth translation).